asciidoc/client-applications.adoc

```diff
@@ -348,377 +324,162 @@ include::{javascript-examples}/examples.test.js[tags=custom-auth]
-----
-======
-====
-
-
-[[driver-configuration]]
-== Configuration
-
-
-[[driver-configuration-encryption]]
-=== Encryption
-
-TLS encryption is enabled for all connections by default.
-This can be disabled through configuration as follows:
-
-[.tabbed-example]
-.Unencrypted
+// [.include-with-python]
+// ======
+// [source, python]
+// ----
+// include::{python-examples}/custom_auth_example.py[tags=custom-auth]
+// ----
+// ======
 ====
-[.include-with-dotnet]
-======
-[source, csharp]
-----
-include::{dotnet-examples}/Examples.cs[tags=config-unencrypted]
-----
-======
-
-[.include-with-go]
-======
-[source, go]
-----
-include::{go-examples}/examples_test.go[tags=config-unencrypted]
-----
-======
-
-[.include-with-java]
-======
-[source, java]
-----
-include::{java-examples}/ConfigUnencryptedExample.java[tags=config-unencrypted]
-----
-======
 
-[.include-with-javascript]
-======
-[source, javascript]
-----
-include::{javascript-examples}/examples.test.js[tags=config-unencrypted]
-----
-======
 
-[.include-with-python]
-======
-[source, python]
-----
-include::{python-examples}/config_unencrypted_example.py[tags=config-unencrypted]
-----
-======
-====
+[[driver-security]]
+== Security
 
-The server can be modified to require encryption for all connections.
-Please see the <<operations-manual#connectors, Operations Manual -> Configure Neo4j connectors>> for more information.
 
-An attempt to connect to a server using an encryption setting not allowed by that server will result in a <<driver-service-unavailable, _Service unavailable_>> status.
+[CAUTION]
+There are significant changes to security settings in Neo4j 4.0. Please read this (and/or consult the migration guide) before upgrading from a previous version.
 
+Starting with Neo4j 4.0, client-server communication uses only *unencrypted local* connections by default.
+This is a change from previous versions, which switched on encryption by default, but generated a self-signed certificate out of the box.
 
-[[driver-configuration-trust]]
-=== Trust
+When a full certificate is installed, and encryption is enabled on the driver, full certificate checks are carried out (refer to <<operations-manual#ssl-framework, Operations Manual -> SSL framework >>).
+Full certificates provide better overall security than self-signed certificates as they include a complete chain of trust back to a root certificate authority.
 
-During a TLS handshake, the server provides a certificate to the client application.
-The application can choose to accept or reject this certificate based on one of the following trust strategies:
+[NOTE]
+https://neo4j.com/aura/[Neo4j Aura] is a secure hosted service backed by full certificates signed by a root certificate authority.
+To connect to Neo4j Aura, driver users must enable encryption and the complete set of certificate checks (the latter of which are enabled by default).
 
-.Trust strategies
+.Default settings
```

Comment:
I suggest (may be worded better, but you get the point):
.Changes in security settings defaults between 3.x and 4.x.

Reply:
Agreed